app/main/posts/views/filters/filters-dropdown.directive.js

```diff
@@ -1,77 +1,75 @@
-            PostFilters.reactiveFilters = true;
-            $scope.canUpdateSavedSearch = false;
+FiltersDropdownController.$inject = ['$scope', '$state', 'PostFilters', 'ModalService', '$rootScope', '_', '$location', 'SavedSearchEndpoint'];
+function FiltersDropdownController($scope, $state, PostFilters, ModalService, $rootScope, _, $location, SavedSearchEndpoint) {
+    $scope.canUpdateSavedSearch = false;
+    PostFilters.reactiveFilters = false;
 
-        };
-        $scope.enableQuery = function () {
-            PostFilters.qEnabled = true;
-        };
-        $scope.saveSavedSearchModal = function () {
-            $scope.savedSearch.filter = $scope.filtersVar;
-            // @TODO Prevent the user from creating one if they somehow manage to get to this point without being logged in
-            $scope.savedSearch.user_id = $rootScope.currentUser ? $rootScope.currentUser.userId : null;
-            ModalService.openTemplate('<saved-search-editor saved-search="savedSearch"></saved-search-editor>', 'set.create_savedsearch', 'star', $scope, false, false);
-        };
-        $scope.editSavedSearchModal = function (editOrUpdate) {
-            let modalHeaderText = editOrUpdate === 'edit' ? 'set.edit_savedsearch' : 'set.update_savedsearch';
+    $scope.$watch(PostFilters.getModeId, function (newValue, oldValue) {
+        if (oldValue !== newValue || typeof ($scope.canUpdateSavedSearch) === 'undefined') {
+            setSavedSearchUpdateStatus();
+        }
+    });
+    // Init an empty saved search
+    $scope.savedSearch = {
+        view : 'map',
+        role : []
+    };
+    // Check if we can edit
+    function setSavedSearchUpdateStatus() {
+        var savedSearch = PostFilters.getModeEntity();
 
-            SavedSearchEndpoint.get({id: PostFilters.getModeId()}, function (savedSearch) {
-                $scope.savedSearch = savedSearch;
-                $scope.savedSearch.filter = PostFilters.getActiveFilters($scope.filtersVar);
-                // @TODO Prevent the user from creating one if they somehow manage to get to this point without being logged in
-                $scope.savedSearch.user_id = $rootScope.currentUser ? $rootScope.currentUser.userId : null;
-                ModalService.openTemplate('<saved-search-editor saved-search="savedSearch"></saved-search-editor>', modalHeaderText, 'star', $scope, false, false);
-            });
-        };
+        $scope.canUpdateSavedSearch = savedSearch && _.contains(savedSearch.allowed_privileges, 'update');
     }
+
+    $scope.applyFiltersLocked = function () {
+        PostFilters.reactiveFilters = true;
+        $scope.dropdownStatus.isopen = !$scope.dropdownStatus.isopen;
+    };
+    $scope.clearFilters = function () {
+        if (PostFilters.getMode() === 'savedsearch' && PostFilters.getModeId()) {
+            PostFilters.setMode('all');
+
+            if ($state.$current.includes['posts.map']) {
```

Comment:
@rowasc so I think this is better than the previous `view` param and setting $location? What do you think?

Reply:
yea, that's way better .